.pre-commit-config.yaml

```diff
@@ -0,0 +1,64 @@
+# .pre-commit-config.yaml
+# for details see https://pre-commit.com
+# for list of available hooks see https://pre-commit.com/hooks.html
+#
+# Preclude commits that do not conform to various criteria.
+#
+# If a pre-commit check must be skipped then use: `SKIP=check_id git commit` Where `check_id` is the id of the check to
+# be skipped such as `black`.
+
+fail_fast: true
+default_stages: [pre-commit]
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v5.0.0
+  hooks:
+   - id: check-added-large-files
+     args: ['--maxkb=200']
+     exclude: |
+       (
+       Source/DataDictionary/gdcmPrivateDefaultDicts.cxx|
+       Documentation/watershed.ps
+       )
+   - id: check-json
+   - id: check-toml
+   - id: check-yaml
+   - id: check-shebang-scripts-are-executable
+   - id: detect-private-key
+   - id: end-of-file-fixer
+     exclude: "\\.(md5|sha|sha512|svg|vtk|vtp)$|^Resources\\/[^\\/]+\\.h$|\\/ColorFiles\\/.+\\.txt$|Data\\/Input\\/.+$"
+   - id: forbid-new-submodules
+   - id: forbid-submodules
+   - id: mixed-line-ending
+     exclude: "\\.(sha|sha512|svg|vtk|vtp)$"
+   - id: name-tests-test
+     args: ['--pytest-test-first']
+   - id: trailing-whitespace
+     exclude: "\\.(sha|sha512|svg|vtk|vtp)$"
+- repo: https://github.com/psf/black
+  rev: 24.2.0
+  hooks: # check conformance to black formatting
+   - id: black
+     args: ['--diff', '-v'] # if run without arguments, will fail and will format the files
+- repo: local
+  hooks:
+    - id: local prepare-commit-msg
+      name: 'local prepare-commit-msg'
+      entry: 'Utilities/Hooks/prepare-commit-msg'
+      language: system
+      stages: [prepare-commit-msg]
+    - id: local pre-commit
```

Comment:
There is not "check-setup-for-development" hook here. It this intentional?

Reply:
Just missed it.  Thank you for identifying.